TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1580,3 +1757,82 @@ def parse_layer_as_float(shape, properties, fid, zoom):
             properties['layer'] = layer_float
 
     return shape, properties, fid
+
+
+def drop_features_where(
+        feature_layers, zoom, source_layer=None, start_zoom=0,
+        property_name=None, drop_property=True,
+        geom_types=None):
+    """
+    Drops some features entirely when they have a property
+    named `property_name` and its value is true. Also can
+    drop the property if `drop_property` is truthy. If
+    `geom_types` is present and not None, then only types in
+    that list are considered for dropping.
+
+    This is useful for dropping features which we want to use
+    earlier in the pipeline (e.g: to generate points), but
+    that we don't want to appear in the final output.
+    """
+
+    assert source_layer, 'drop_features_where: missing source layer'
+    assert property_name, 'drop_features_where: missing property name'
+
+    if zoom < start_zoom:
+        return None
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    new_features = []
+    for feature in layer['features']:
+        shape, properties, fid = feature
+
+        matches_geom_type = \
+            geom_types is None or \
+            shape.geom_type in geom_types
+
+        val = None
+        if drop_property:
+            val = properties.pop(property_name, None)
+        else:
+            val = properties.get(property_name)
+
+        if val == True and matches_geom_type:
+            pass
+        else:
+            new_features.append((shape, properties, fid))
+
+    layer['features'] = new_features
+    return layer
+
+
+def remove_zero_area(shape, properties, fid, zoom):
+    """
+    All features get a numeric area tag, but for points this
+    is zero. The area probably isn't exactly zero, so it's
+    probably less confusing to just remove the tag to show
+    that the value is probably closer to "unspecified".
+    """
+
+    # remove the property if it's present. we _only_ want
+    # to replace it if it matches the positive, float
+    # criteria.
+    area = properties.pop("area", None)
+
+    # try to parse a string if the area has been sent as a
+    # string. it should come through as a float, though,
+    # since postgres treats it as a real.
+    if isinstance(area, str):
+        area = to_float(area)
+
+    if area is not None:
+        # cast to integer to match what we do for polygons.
+        # also the fractional parts of a sq.m are just
+        # noise really.
+        area = int(area)
+        if area > 0:
+            properties['area'] = area
```

Comment:
Area (a.k.a. `way_area`) generally comes through as a `float`, since that column has `real` type in Postgres. The only reason I added the `to_float` was in case that changed in the future, and left us unexpectedly throwing exceptions from the `int()` conversion.